temporal-testing/src/main/java/io/temporal/internal/testservice/TestWorkflowStoreImpl.java

```diff
@@ -288,6 +293,22 @@ public void registerDelayedCallback(Duration delay, Runnable r) {
     timerService.schedule(delay, r, "registerDelayedCallback");
   }
 
+  @Override
+  public void registerSearchAttribute(String name, Class<?> type) {
+    SearchAttributeType encodedType = javaTypeToEncodedType(type);
+    if (encodedType == SearchAttributeType.Unspecified) {
+      throw Status.INVALID_ARGUMENT
+          .withDescription("Unable to read search attribute type: " + type)
+          .asRuntimeException();
+    }
+    searchAttributes.put(name, new Pair(null, encodedType));
```

Comment:
Pair belongs to this class. It's the only class that creates, persists and extracts them. All other classes are only passing existing entities around.